src/extensions/default/bramble/lib/RemoteEvents.js

```diff
@@ -157,6 +157,40 @@ define(function (require, exports, module) {
             });
         });
 
+        // Listen for changes to TagHints
+        PreferencesManager.on("change", "codehint.TagHints", function () {
```

Comment:
I don't like the idea of sending 4 separate events for state changes on a variable that we're tracking in the StateManager as a single boolean.  Can you harmonize these so they all send the same `type: "bramble:autocompleteChange"` with a single boolean?  It will mean we fire this change event 4 times, but that shouldn't break anything, since Thimble will just set it `true` or `false` in all cases.